WcaOnRails/app/webpacker/components/EditEvents/EventPanel/RoundRow.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+
+import {
+  Dropdown, Input, Table,
+} from 'semantic-ui-react';
+import { events, formats } from '../../../lib/wca-data.js.erb';
+import { roundIdToString } from '../../../lib/utils/wcif';
+
+import {
+  EditAdvancementConditionModal, EditTimeLimitModal, EditCutoffModal,
+} from '../Modals';
+import { useDispatch } from '../../../lib/providers/StoreProvider';
+import { useConfirm } from '../../../lib/providers/ConfirmProvider';
+import { updateRoundFormat, setScrambleSetCount, updateCutoff } from '../store/actions';
+
+export default function RoundRow({
+  index, wcifRound, wcifEvent, disabled,
+}) {
+  const dispatch = useDispatch();
+  const confirm = useConfirm();
+  const event = events.byId[wcifEvent.id];
+
+  const roundNumber = index + 1;
+  const isLastRound = roundNumber === wcifEvent.rounds.length;
+
+  const roundFormatChanged = (e, { value }) => {
+    const newFormat = value;
+
+    if (
+      wcifRound.cutoff
+      && !formats.byId[newFormat].allowedFirstPhaseFormats.includes(
+        wcifRound.cutoff.numberOfAttempts.toString(),
+      )
+    ) {
+      // if the format is changing to a format that doesn't have a cutoff
+      confirm({
+        content: `Are you sure you want to change the format of ${roundIdToString(wcifRound.id)}? This will clear the cutoff`,
+      })
+        .then(() => {
+          dispatch(updateRoundFormat(wcifRound.id, newFormat));
+          dispatch(updateCutoff(wcifRound.id, null));
+        });
+    } else {
+      // if the format is changing to a format that has a cutoff
+      dispatch(updateRoundFormat(wcifRound.id, newFormat));
+    }
+  };
+
+  const scrambleSetCountChanged = (e) => {
+    dispatch(setScrambleSetCount(wcifRound.id, parseInt(e.target.value, 10))); // XXX
```

Comment:
parseInt with 10 specified as the base is a standard certain eslint plugins like to see.
I don't think this needs to be changed.
As for the XXX, it might've been meant to communicate to myself that I planned to revisit this line of code but I really should've included a message. With no message, I have no way of knowing what it meant but if things work, we can likely remove the comment.